core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/scheduling/ExpansionGreedyRegionPlanGenerator.scala

```diff
@@ -100,7 +104,19 @@ class ExpansionGreedyRegionPlanGenerator(
             physicalPlan.getUpstreamPhysicalLinkIds(operatorId) ++ physicalPlan
               .getDownstreamPhysicalLinkIds(operatorId)
           })
-          Region(RegionIdentity((index + 1).toString), operatorIds, linkIds)
+          val region = Region(RegionIdentity((index + 1).toString), operatorIds, linkIds)
+
+          val resourceAllocation = new ResourceAllocation(region)
+          val executionClusterInfo = new ExecutionClusterInfo()
+          val (workers, _) = resourceAllocation.allocation(physicalPlan, executionClusterInfo)
```

Comment:
lets use one resource allocator for all regions.
also the cluster information can be one copy for all regions as well.